File: 2024/python/template.py
def get_data(path):
    with open(path) as file:
        data = file.read().split("\n")

    return data


def part_one():
    _ = get_data("smallInput.txt")


def part_two():
    _ = get_data("smallInput.txt")


if __name__ == "__main__":
    part_one()
    part_two()

File: 2024/haskell/template.hs
-- Function to read and parse the input file into two arrays of numbers
parseInput :: FilePath -> IO --TODO: Insert output type
parseInput filename = do
    content <- readFile filename
    let linesOfFile = lines content


-- Function to read smallInput.txt specifically
readInput :: IO --TODO: Insert output type
readInput = parseInput "input.txt"


-- Example usage function
partOne :: IO ()
partOne = do
    input <- readInput


partTwo :: IO ()
partTwo = do
    input <- readInput


main :: IO ()
main = do
    partOne
    partTwo